### DISCUSSION TITLE
About \dot {q} in Pinocchio

### DISCUSSION THREAD

**Discussion by Czworldy** [stack-of-tasks/pinocchio#2248] [Q&A] [answered]
My question is very simple, 
we have a float-based robot, and we add the floating-base to `pinocchio` model by the code
```
// add 6 DoF for the floating base
pinocchio::JointModelComposite jointComposite(2);
jointComposite.addJoint(pinocchio::JointModelTranslation());
jointComposite.addJoint(pinocchio::JointModelSphericalZYX());
pinocchio::ModelTpl<double> model;
pinocchio::urdf::buildModel(urdfTree, jointComposite, model);
```
and when call `pinocchio::forwardKinematics(model, data, qPinocchio, vPinocchio); ` or `pinocchio::rnea(mode, data, qPinocchio, vPinocchio, aPinocchio......); `functions, we konw that the `qPinocchio.head<6>()` should be `[x, y, z,   z-,y-,x-euler angles]`
but how about `qPinocchio` and `aPinocchio`? Should we use **Euler Angles Zyx (second) Derivatives** or **Global Angular Velocity /Acceleration**?
Thanks a lot!

**Comment by ManifoldFR** [comment, marked as answer]
Hello ! Thanks for your interest in Pinocchio.

If you have a floating-base robot defined using the free-flyer joint model, then the corresponding part of the $q$ configuration vector is actually the _xyz_ for the base position followed by the 4D quaternion corresponding the orientation of the robot (so, not the Euler angles). Notably, this means the part of $q$ you want is actually `qPinocchio.head<7>()`. Pinocchio does not use Euler angles, but the appropriate double-covering quaternion representation (although you can convert between these representations using tools provided in the library).

For the velocity and acceleration vectors $v$ you'll get the 6D $(v_x,v_y,v_z,\omega_x,\omega_y,\omega_z)$ spatial velocity from `vPinocchio.head<6>()`.

**Reply by Czworldy** [reply]
thanks for your kind reply. 
if i use `pinocchio::JointModelSphericalZYX()` in pinocchio 3 but i find this bug still exists.
> https://github.com/stack-of-tasks/pinocchio/issues/2053
Can I assume that Pinocchio still does not support `Pinocchio:: JointModelSphereZYX()`, as this may lead to some calculation errors
